### PR TITLE
Update run time to v0.9.3

### PR DESCRIPTION
**Pull request by terencechain**
part of #4098

### CODE REVIEW

File: validator/client/validator_propose.go [hunk 1]
```diff
@@ -85,9 +86,17 @@ func (v *validator) ProposeBlock(ctx context.Context, slot uint64, pubKey [48]by
 	}).Info("Submitted new block")
 }
 
+// ProposeExit --
```
Comment (0xKiwi) [inline validator/client/validator_propose.go]: Is there an issue we can reference and comment here?

File: shared/testutil/deposits.go [hunk 2]
```diff
@@ -189,3 +189,79 @@ func ResetCache() {
 	privKeys = []*bls.SecretKey{}
 	cachedDeposits = []*ethpb.Deposit{}
 }
+
+// DeterministicDepositsAndKeysSameValidator returns the entered amount of deposits and secret keys.
```
Comment (0xKiwi) [inline shared/testutil/deposits.go]:
```suggestion
// DeterministicDepositsAndKeysSameValidator returns the entered amount of deposits and secret keys
```

File: shared/featureconfig/flags.go [hunk 3]
```diff
@@ -142,6 +139,11 @@ var (
 		Usage: deprecatedUsage,
 		Hidden: true,
 	}
+	deprecatedGenesisDelayFlag = cli.BoolFlag{
```
Comment (0xKiwi) [inline shared/featureconfig/flags.go]: How long do we keep a flag deprecated?

Comment (prestonvanloon) [reply shared/featureconfig/flags.go]: No later than v1 release.

File: proto/testing/ssz_regression_test.go [hunk 4]
```diff
@@ -53,7 +54,7 @@ func TestBlockHeaderSigningRoot(t *testing.T) {
 				ParentRoot: hexDecodeOrDie(t, "f9b2785de53069d4ad16cc0ec729afe9f879e391433ec120bb15b5082a486705"),
 				StateRoot:  hexDecodeOrDie(t, "737d1c6ff6e2edf7f0627bf55381e6b08f6c2c56ed8d1895ae47a782dc09382e"),
 				BodyRoot:   hexDecodeOrDie(t, "affff5006c34a3a2bf7f18b7860675f002187ea809f708fa8f44c424321bcd1c"),
-				Signature:  hexDecodeOrDie(t, "17d25044259a0ccd99d1b45eeec4e084e5fb0fef98d5805001b248feb555b947ecf6842b9ad546f98f63ef89117575d73223e9fb9ee8143857b6fcc79600fffed1966cea46f7524236cd1e83531aef906cb8b4c296d50695bb83efa84075d309"),
+				//Signature:  hexDecodeOrDie(t, "17d25044259a0ccd99d1b45eeec4e084e5fb0fef98d5805001b248feb555b947ecf6842b9ad546f98f63ef89117575d73223e9fb9ee8143857b6fcc79600fffed1966cea46f7524236cd1e83531aef906cb8b4c296d50695bb83efa84075d309"),
```
Comment (0xKiwi) [inline proto/testing/ssz_regression_test.go]:
```suggestion
```

File: beacon-chain/core/helpers/committee.go [hunk 5]
```diff
@@ -356,7 +356,7 @@ func VerifyBitfieldLength(bf bitfield.Bitfield, committeeSize uint64) error {
 	return nil
 }
 
-// VerifyAttestationBitfieldLengths verifies that an attestations aggregation and custody bitfields are
+// VerifyAttestationBitfieldLengths verifies that an attestations aggregation bitfields is
```
Comment (0xKiwi) [inline beacon-chain/core/helpers/committee.go]:
```suggestion
// VerifyAttestationBitfieldLengths verifies that an attestations aggregation bitfield is
```

File: beacon-chain/core/epoch/epoch_processing.go [hunk 6]
```diff
@@ -160,18 +170,16 @@ func ProcessRegistryUpdates(state *pb.BeaconState) (*pb.BeaconState, error) {
 		}
 	}
 
-	// Queue the validators whose eligible to activate and sort them by activation eligibility epoch number
+	// Queue validators eligible for activation and not yet dequeued for activation
```
Comment (0xKiwi) [inline beacon-chain/core/epoch/epoch_processing.go]:
```suggestion
	// Queue validators eligible for activation and not yet dequeued for activation.
```

File: beacon-chain/cache/depositcache/pending_deposits.go [hunk 7]
```diff
@@ -5,6 +5,8 @@ import (
 	"math/big"
 	"sort"
 
+	dbpb "github.com/prysmaticlabs/prysm/proto/beacon/db"
+
```
Comment (0xKiwi) [inline beacon-chain/cache/depositcache/pending_deposits.go]:
```suggestion
```

File: beacon-chain/operations/attestations/kv/forkchoice.go [hunk 8]
```diff
@@ -0,0 +1,61 @@
...
+		}
+		atts = append(atts, att)
+	}
+
+	return atts
+}
+
+// DeleteForkchoiceAttestation deletes a forkchoice attestation in cache.
+func (p *AttCaches) DeleteForkchoiceAttestation(att *ethpb.Attestation) error {
+
```
Comment (0xKiwi) [inline beacon-chain/operations/attestations/kv/forkchoice.go]:
```suggestion
```

File: beacon-chain/operations/attestations/pool.go [hunk 9]
```diff
@@ -5,24 +5,35 @@ import (
...
 	DeleteAggregatedAttestation(att *ethpb.Attestation) error
 	// For unaggregated attestations
 	SaveUnaggregatedAttestation(att *ethpb.Attestation) error
-	UnaggregatedAttestation(slot uint64, committeeIndex uint64) []*ethpb.Attestation
+	SaveUnaggregatedAttestations(atts []*ethpb.Attestation) error
+	UnaggregatedAttestations() []*ethpb.Attestation
 	DeleteUnaggregatedAttestation(att *ethpb.Attestation) error
-	// For forkchoice attestations
-
+	// For attestations that were included in the block
```
Comment (0xKiwi) [inline beacon-chain/operations/attestations/pool.go]:
```suggestion
	// For attestations that were included in the block.
```

File: beacon-chain/operations/attestations/pool.go [hunk 10]
```diff
@@ -5,24 +5,35 @@ import (
...
-type AttestationPool interface {
+type Pool interface {
 	// For Aggregated attestations
 	SaveAggregatedAttestation(att *ethpb.Attestation) error
-	AggregatedAttestation() []*ethpb.Attestation
+	SaveAggregatedAttestations(atts []*ethpb.Attestation) error
+	AggregatedAttestations() []*ethpb.Attestation
+	AggregatedAttestationsBySlotIndex(slot uint64, committeeIndex uint64) []*ethpb.Attestation
 	DeleteAggregatedAttestation(att *ethpb.Attestation) error
 	// For unaggregated attestations
```
Comment (0xKiwi) [inline beacon-chain/operations/attestations/pool.go]:
```suggestion
	// For unaggregated attestations.
```

File: beacon-chain/operations/attestations/pool.go [hunk 11]
```diff
@@ -5,24 +5,35 @@ import (
...
+	UnaggregatedAttestations() []*ethpb.Attestation
 	DeleteUnaggregatedAttestation(att *ethpb.Attestation) error
-	// For forkchoice attestations
-
+	// For attestations that were included in the block
+	SaveBlockAttestation(att *ethpb.Attestation) error
+	SaveBlockAttestations(atts []*ethpb.Attestation) error
+	BlockAttestations() []*ethpb.Attestation
+	DeleteBlockAttestation(att *ethpb.Attestation) error
+	// For attestations to be passed to fork choice
```
Comment (0xKiwi) [inline beacon-chain/operations/attestations/pool.go]:
```suggestion
	// For attestations to be passed to fork choice.
```

File: beacon-chain/operations/attestations/prepare_forkchoice.go [hunk 12]
```diff
@@ -0,0 +1,106 @@
...
+
+	ethpb "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
+	"github.com/prysmaticlabs/go-ssz"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"go.opencensus.io/trace"
+)
+
+// prepare attestations for fork choice at every half of the slot.
```
Comment (0xKiwi) [inline beacon-chain/operations/attestations/prepare_forkchoice.go]:
```suggestion
// Prepare attestations for fork choice at every half of the slot.
```

File: beacon-chain/operations/attestations/service.go [hunk 13]
```diff
@@ -0,0 +1,65 @@
...
+
+import (
+	"context"
+
+	"github.com/dgraph-io/ristretto"
+)
+
+var forkChoiceProcessedRootsSize = int64(1 << 16)
+
+// Service of attestation pool operations
```
Comment (0xKiwi) [inline beacon-chain/operations/attestations/service.go]:
```suggestion
// Service of attestation pool operations.
```

File: beacon-chain/operations/attestations/service_test.go [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+		t.Fatalf("Unable to stop attestation pool service: %v", err)
+	}
+
+	if s.ctx.Err() != context.Canceled {
+		t.Error("context was not canceled")
+	}
+}
+
+func TestStatus_Error(t *testing.T) {
+	err := errors.New("bad bad bad")
```
Comment (0xKiwi) [inline beacon-chain/operations/attestations/service_test.go]: Is this an error message that gets logged out?

File: beacon-chain/powchain/log_processing_test.go [hunk 15]
```diff
@@ -604,3 +629,95 @@ func TestWeb3ServiceProcessDepositLog_RequestMissedDeposits(t *testing.T) {
...
+
+		testAcc.Backend.Commit()
+	}
+
+	time.Sleep(2 * time.Second)
+	if !web3Service.chainStartData.Chainstarted {
+		t.Fatalf("Service hasn't chainstarted yet with a block height of %d", web3Service.latestEth1Data.BlockHeight)
+	}
+
+	// Advance 10 blocks
```
Comment (0xKiwi) [inline beacon-chain/powchain/log_processing_test.go]:
```suggestion
	// Advance 10 blocks.
```

File: beacon-chain/powchain/log_processing_test.go [hunk 16]
```diff
@@ -604,3 +629,95 @@ func TestWeb3ServiceProcessDepositLog_RequestMissedDeposits(t *testing.T) {
...
+	for i := 0; i < 10; i++ {
+		testAcc.Backend.Commit()
+	}
+
+	// tearing down to prevent registration error
```
Comment (0xKiwi) [inline beacon-chain/powchain/log_processing_test.go]:
```suggestion
	// Tearing down to prevent registration error.
```

File: beacon-chain/powchain/log_processing_test.go [hunk 17]
```diff
@@ -604,3 +629,95 @@ func TestWeb3ServiceProcessDepositLog_RequestMissedDeposits(t *testing.T) {
...
+	_, roots, err := testutil.DeterministicDepositTrie(len(deposits))
+	if err != nil {
+		t.Fatal(err)
+	}
+	ctx, cancel := context.WithCancel(context.Background())
+	go web3Service.run(ctx.Done())
+
+	// 64 Validators are used as size required for beacon-chain to start. This number
+	// is defined in the deposit contract as the number required for the testnet. The actual number
+	// is 2**14
```
Comment (0xKiwi) [inline beacon-chain/powchain/log_processing_test.go]:
```suggestion
	// is 2**14.
```

File: beacon-chain/rpc/aggregator/server.go [hunk 18]
```diff
@@ -70,13 +75,30 @@ func (as *Server) SubmitAggregateAndProof(ctx context.Context, req *pb.Aggregati
 		return nil, status.Errorf(codes.InvalidArgument, "Validator is not an aggregator")
 	}
 
-	// TODO(3865): Broadcast aggregated attestation & proof via the aggregation topic
+	// Retrieve the unaggregated attestation from pool
```
Comment (0xKiwi) [inline beacon-chain/rpc/aggregator/server.go]:
```suggestion
	// Retrieve the unaggregated attestation from pool.
```

File: beacon-chain/rpc/testing/beacon_chain_service_mock.go [hunk 19]
```diff
@@ -1,16 +1,15 @@
...
 package testing
 
 import (
 	context "context"
-	reflect "reflect"
-
 	gomock "github.com/golang/mock/gomock"
 	v1alpha1 "github.com/prysmaticlabs/ethereumapis/eth/v1alpha1"
 	metadata "google.golang.org/grpc/metadata"
+	reflect "reflect"
```
Comment (0xKiwi) [inline beacon-chain/rpc/testing/beacon_chain_service_mock.go]: Not sorted properly, maybe call goimports?

File: beacon-chain/rpc/validator/exit.go [hunk 20]
```diff
@@ -0,0 +1,37 @@
...
+)
+
+// ProposeExit proposes an exit for a validator.
+func (vs *Server) ProposeExit(ctx context.Context, req *ethpb.SignedVoluntaryExit) (*ptypes.Empty, error) {
+	s, err := vs.HeadFetcher.HeadState(ctx)
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Could not get head state: %v", err)
+	}
+
+	// Confirm the validator is eligible to exit with the parameters provided
```
Comment (0xKiwi) [inline beacon-chain/rpc/validator/exit.go]:
```suggestion
	// Confirm the validator is eligible to exit with the parameters provided.
```

File: beacon-chain/rpc/validator/exit_test.go [hunk 21]
```diff
@@ -0,0 +1,93 @@
...
+	opfeed "github.com/prysmaticlabs/prysm/beacon-chain/core/feed/operation"
+	"github.com/prysmaticlabs/prysm/beacon-chain/core/state"
+	dbutil "github.com/prysmaticlabs/prysm/beacon-chain/db/testing"
+	mockSync "github.com/prysmaticlabs/prysm/beacon-chain/sync/initial-sync/testing"
+	"github.com/prysmaticlabs/prysm/shared/params"
+	"github.com/prysmaticlabs/prysm/shared/testutil"
+)
+
+func TestSub(t *testing.T) {
+
```
Comment (0xKiwi) [inline beacon-chain/rpc/validator/exit_test.go]:
```suggestion
```

File: beacon-chain/rpc/validator/exit_test.go [hunk 22]
```diff
@@ -0,0 +1,93 @@
...
+		},
+		Signature: []byte{0xb3, 0xe1, 0x9d, 0xc6, 0x7c, 0x78, 0x6c, 0xcf, 0x33, 0x1d, 0xb9, 0x6f, 0x59, 0x64, 0x44, 0xe1, 0x29, 0xd0, 0x87, 0x03, 0x26, 0x6e, 0x49, 0x1c, 0x05, 0xae, 0x16, 0x7b, 0x04, 0x0f, 0x3f, 0xf8, 0x82, 0x77, 0x60, 0xfc, 0xcf, 0x2f, 0x59, 0xc7, 0x40, 0x0b, 0x2c, 0xa9, 0x23, 0x8a, 0x6c, 0x8d, 0x01, 0x21, 0x5e, 0xa8, 0xac, 0x36, 0x70, 0x31, 0xb0, 0xe1, 0xa8, 0xb8, 0x8f, 0x93, 0x8c, 0x1c, 0xa2, 0x86, 0xe7, 0x22, 0x00, 0x6a, 0x7d, 0x36, 0xc0, 0x2b, 0x86, 0x2c, 0xf5, 0xf9, 0x10, 0xb9, 0xf2, 0xbd, 0x5e, 0xa6, 0x5f, 0x12, 0x86, 0x43, 0x20, 0x4d, 0xa2, 0x9d, 0x8b, 0xe6, 0x6f, 0x09},
+	}
+
+	_, err = server.ProposeExit(context.Background(), req)
+	if err != nil {
+		t.Fatalf("Unexpected error: %v", err)
+	}
+
+	// Ensure the state notification was broadcast
```
Comment (0xKiwi) [inline beacon-chain/rpc/validator/exit_test.go]:
```suggestion
	// Ensure the state notification was broadcast.
```

File: beacon-chain/rpc/validator/exit_test.go [hunk 23]
```diff
@@ -0,0 +1,93 @@
...
+		StateNotifier:     mockChainService.StateNotifier(),
+		OperationNotifier: mockChainService.OperationNotifier(),
+	}
+
+	// Subscribe to operation notifications
+	opChannel := make(chan *feed.Event, 1024)
+	opSub := server.OperationNotifier.OperationFeed().Subscribe(opChannel)
+	defer opSub.Unsubscribe()
+
+	// Send the request, expect a result on the state feed
```
Comment (0xKiwi) [inline beacon-chain/rpc/validator/exit_test.go]:
```suggestion
	// Send the request, expect a result on the state feed.
```

File: beacon-chain/rpc/validator/exit_test.go [hunk 24]
```diff
@@ -0,0 +1,93 @@
...
+	server := &Server{
+		BeaconDB:          db,
+		HeadFetcher:       mockChainService,
+		SyncChecker:       &mockSync.Sync{IsSyncing: false},
+		GenesisTime:       genesisTime,
+		StateNotifier:     mockChainService.StateNotifier(),
+		OperationNotifier: mockChainService.OperationNotifier(),
+	}
+
+	// Subscribe to operation notifications
```
Comment (0xKiwi) [inline beacon-chain/rpc/validator/exit_test.go]:
```suggestion
	// Subscribe to operation notifications.
```

File: beacon-chain/rpc/validator/exit_test.go [hunk 25]
```diff
@@ -0,0 +1,93 @@
...
+	block := blk.NewGenesisBlock([]byte{})
+	if err := db.SaveBlock(ctx, block); err != nil {
+		t.Fatalf("Could not save genesis block: %v", err)
+	}
+	genesisRoot, err := ssz.HashTreeRoot(block.Block)
+	if err != nil {
+		t.Fatalf("Could not get signing root %v", err)
+	}
+
+	// Set genesis time to be 100 epochs ago
```
Comment (0xKiwi) [inline beacon-chain/rpc/validator/exit_test.go]:
```suggestion
	// Set genesis time to be 100 epochs ago.
```

File: beacon-chain/rpc/validator/proposer.go [hunk 26]
```diff
@@ -1,101 +1,70 @@
...
 	ctx, span := trace.StartSpan(ctx, "ProposerServer.RequestBlock")
 	defer span.End()
 	span.AddAttributes(trace.Int64Attribute("slot", int64(req.Slot)))
 
-	if ps.SyncChecker.Syncing() {
+	if vs.SyncChecker.Syncing() {
 		return nil, status.Errorf(codes.Unavailable, "Syncing to latest head, not ready to respond")
 	}
 
 	// Retrieve the parent block as the current head of the canonical chain
```
Comment (0xKiwi) [inline beacon-chain/rpc/validator/proposer.go]:
```suggestion
	// Retrieve the parent block as the current head of the canonical chain.
```

File: beacon-chain/rpc/validator/proposer_test.go [hunk 27]
```diff
@@ -1,21 +1,27 @@
-package proposer
+package validator
 
 import (
 	"context"
+	dbpb "github.com/prysmaticlabs/prysm/proto/beacon/db"
+	"github.com/prysmaticlabs/prysm/shared/stateutil"
```
Comment (0xKiwi) [inline beacon-chain/rpc/validator/proposer_test.go]: Sorting is wrong here.

File: beacon-chain/sync/subscriber.go [hunk 28]
```diff
@@ -150,26 +157,23 @@ func (r *Service) subscribe(topic string, validator pubsub.Validator, handle sub
 		for {
 			msg, err := sub.Next(r.ctx)
 			if err != nil {
+				// This should only happen when the context is cancelled or subscription is cancelled?.
```
Comment (0xKiwi) [inline beacon-chain/sync/subscriber.go]: Should this be removed?.

File: beacon-chain/sync/subscriber.go [hunk 29]
```diff
@@ -184,3 +188,46 @@ func wrapAndReportValidation(topic string, v pubsub.Validator) (string, pubsub.V
...
+	stateChannel := make(chan *feed.Event, 1)
+	stateSub := r.stateNotifier.StateFeed().Subscribe(stateChannel)
+	go func() {
+		for {
+			select {
+			case <-r.ctx.Done():
+				stateSub.Unsubscribe()
+				return
+			case <-stateChannel:
+				// Update topic count
```
Comment (0xKiwi) [inline beacon-chain/sync/subscriber.go]:
```suggestion
				// Update topic count.
```

File: beacon-chain/sync/subscriber_beacon_blocks_test.go [hunk 30]
```diff
@@ -26,26 +29,61 @@ func TestRegularSyncBeaconBlockSubscriber_FilterByFinalizedEpoch(t *testing.T) {
...
+		t.Fatal(err)
+	}
+	if err := r.attPool.SaveAggregatedAttestation(att3); err != nil {
+		t.Fatal(err)
+	}
+	if err := r.attPool.SaveUnaggregatedAttestation(att4); err != nil {
+		t.Fatal(err)
+	}
+
+	// Seen 1, 3 and 4 in block
```
Comment (0xKiwi) [inline beacon-chain/sync/subscriber_beacon_blocks_test.go]:
```suggestion
	// Seen 1, 3 and 4 in block.
```

File: beacon-chain/sync/subscriber_beacon_blocks_test.go [hunk 31]
```diff
@@ -26,26 +29,61 @@ func TestRegularSyncBeaconBlockSubscriber_FilterByFinalizedEpoch(t *testing.T) {
...
+	if err := r.deleteAttsInPool([]*ethpb.Attestation{att1, att3, att4}); err != nil {
+		t.Fatal(err)
+	}
+
+	// Only 2 should remain
```
Comment (0xKiwi) [inline beacon-chain/sync/subscriber_beacon_blocks_test.go]:
```suggestion
	// Only 2 should remain.
```

File: beacon-chain/rpc/validator/proposer_test.go (hunk 27 above)
Comment (terencechain) [reply beacon-chain/rpc/validator/proposer_test.go]: gofmt works for me

File: beacon-chain/core/state/state.go [hunk 32]
```diff
@@ -51,10 +52,54 @@ import (
 //	    state.active_index_roots[index] = active_index_root
 //	    state.compact_committees_roots[index] = committee_root
 //	  return state
+// This method differs from the spec so as to process deposits beforehand instead of the end of the function.
 func GenesisBeaconState(deposits []*ethpb.Deposit, genesisTime uint64, eth1Data *ethpb.Eth1Data) (*pb.BeaconState, error) {
```
Comment (rauljordan) [inline beacon-chain/core/state/state.go]: Is there a plan to deprecate this in the future or maybe delete this? It's only really used in tests and don't like the pattern of having duplicated functions with slightly different functionality

File: beacon-chain/operations/attestations/aggregate.go [hunk 33]
```diff
@@ -0,0 +1,77 @@
...
+)
+
+// Define time to aggregate the unaggregated attestations at 3 times per slot, this gives
+// enough confidence all the unaggregated attestations will be aggregated as aggregator requests.
+var timeToAggregate = time.Duration(params.BeaconConfig().SecondsPerSlot/3) * time.Second
+
+// This kicks off a routine to aggregate the unaggregated attestations from pool.
+func (s *Service) aggregateRoutine() {
+	ticker := time.NewTicker(timeToAggregate)
+	ctx := context.TODO()
```
Comment (rauljordan) [inline beacon-chain/operations/attestations/aggregate.go]: why is this context todo?

File: beacon-chain/rpc/validator/proposer.go [hunk 34]
```diff
@@ -1,101 +1,70 @@
...
+package validator
 
 import (
 	"context"
 	"fmt"
 	"math/big"
 	"math/rand"
 
+	dbpb "github.com/prysmaticlabs/prysm/proto/beacon/db"
+
```
Comment (rauljordan) [inline beacon-chain/rpc/validator/proposer.go]:
```suggestion
```

File: shared/stateutil/state_root.go [hunk 35]
```diff
@@ -24,7 +24,10 @@ func init() {
 		// 100,000 roots will take up approximately 3 MB in memory.
 		BufferItems: 64, // number of keys per Get buffer.
 	})
-	globalHasher = &stateRootHasher{rootsCache: rootsCache}
+	// Temporarily disable roots cache until cache issues can be resolved.
```
Comment (rauljordan) [inline shared/stateutil/state_root.go]: Need to open a tracking issue for this

File: validator/client/service.go [hunk 36]
```diff
@@ -73,6 +73,9 @@ func (v *ValidatorService) Start() {
 	}
 	opts := []grpc.DialOption{
 		dialOpt,
+		grpc.WithDefaultCallOptions(
+			grpc.MaxCallRecvMsgSize(10 * 5 << 20), // 10Mb
```
Comment (rauljordan) [inline validator/client/service.go]: Should be configurable, maybe open a tracking issue for this

File: beacon-chain/sync/validate_aggregate_proof.go [hunk 37]
```diff
@@ -0,0 +1,169 @@
...
+	}
+	m, ok := raw.(*ethpb.AggregateAttestationAndProof)
+	if !ok {
+		return false
+	}
+
+	attSlot := m.Aggregate.Data.Slot
+
+	// Verify aggregate attestation has not already been seen via aggregate gossip, within a block, or through the creation locally.
+	// TODO(3835): Blocked by operation pool redesign
```
Comment (rauljordan) [inline beacon-chain/sync/validate_aggregate_proof.go]: Maybe we can open a specific tracking issue for this item instead of just relying on #3835

File: validator/client/service.go (hunk 36 above)
Comment (rauljordan) [reply validator/client/service.go]: See #4444

File: shared/stateutil/state_root.go (hunk 35 above)
Comment (rauljordan) [reply shared/stateutil/state_root.go]: Opened #4445

File: beacon-chain/core/state/state.go (hunk 32 above)
Comment (rauljordan) [reply beacon-chain/core/state/state.go]: See #4446

File: beacon-chain/sync/validate_aggregate_proof.go (hunk 37 above)
Comment (terencechain) [reply beacon-chain/sync/validate_aggregate_proof.go]: I'll address this today

File: beacon-chain/blockchain/receive_block.go [hunk 38]
```diff
@@ -218,35 +217,18 @@ func (s *Service) ReceiveBlockNoVerify(ctx context.Context, block *ethpb.BeaconB
...
 	}).Debug("Finished applying state transition")
 
 	return nil
 }
 
-// cleanupBlockOperations processes and cleans up any block operations relevant to the beacon node
-// such as attestations, exits, and deposits. We update the latest seen attestation by validator
-// in the local node's runtime, cleanup and remove pending deposits which have been included in the block
-// from our node's local cache, and process validator exits and more.
-func (s *Service) cleanupBlockOperations(ctx context.Context, block *ethpb.BeaconBlock) error {
```
Comment (prestonvanloon) [inline beacon-chain/blockchain/receive_block.go]: Why was this removed?
